crates/bevy_ecs/src/query/access.rs

```diff
@@ -54,76 +56,115 @@ impl<T: SparseSetIndex> Access<T> {
+    /// Returns `true` if the access can write the element given by `index`.
     pub fn has_write(&self, index: T) -> bool {
-        self.writes.contains(index.sparse_set_index())
+        if self.writes_all {
+            true
+        } else {
+            self.writes.contains(index.sparse_set_index())
+        }
     }
 
-    /// Sets this `Access` to having read access for all indices.
+    /// Sets this `Access` as having read access to all indexed elements.
     pub fn read_all(&mut self) {
         self.reads_all = true;
     }
 
-    /// Returns true if this `Access` has read access to all indices.
-    pub fn reads_all(&self) -> bool {
+    /// Returns `true` if the access can read all elements in the collection.
+    #[inline]
+    pub fn has_read_all(&self) -> bool {
         self.reads_all
     }
 
-    /// Clears all recorded accesses.
+    /// Sets this `Access` as having write access to all indexed elements.
+    pub fn write_all(&mut self) {
+        self.reads_all = true;
+        self.writes_all = true;
+    }
+
+    /// Returns `true` if the access can write all elements in the collection.
+    #[inline]
+    pub fn has_write_all(&self) -> bool {
+        self.writes_all
+    }
+
+    /// Removes all accesses.
     pub fn clear(&mut self) {
         self.reads_all = false;
+        self.writes_all = false;
         self.reads_and_writes.clear();
         self.writes.clear();
     }
 
-    /// Extends this `Access` with another, copying all accesses of `other` into this.
+    /// Adds all read and write access from `other`.
     pub fn extend(&mut self, other: &Access<T>) {
         self.reads_all = self.reads_all || other.reads_all;
+        self.writes_all = self.writes_all || other.writes_all;
         self.reads_and_writes.union_with(&other.reads_and_writes);
         self.writes.union_with(&other.writes);
     }
 
-    /// Returns true if this `Access` is compatible with `other`.
+    /// Returns `true` if the access and `other` can be active at the same time.
     ///
-    /// Two `Access` instances are incompatible with each other if one `Access` has a write for
-    /// which the other also has a write or a read.
+    /// `Access` instances are incompatible if one can write
+    /// an element that the other can read or write.
     pub fn is_compatible(&self, other: &Access<T>) -> bool {
-        if self.reads_all {
-            0 == other.writes.count_ones(..)
+        if self.writes_all {
+            other.reads_and_writes.count_ones(..) == 0
+        } else if other.writes_all {
+            self.reads_and_writes.count_ones(..) == 0
+        } else if self.reads_all {
+            other.writes.count_ones(..) == 0
         } else if other.reads_all {
-            0 == self.writes.count_ones(..)
+            self.writes.count_ones(..) == 0
         } else {
             self.writes.is_disjoint(&other.reads_and_writes)
                 && self.reads_and_writes.is_disjoint(&other.writes)
         }
     }
```

Comment:
not super familiar with this code but I think this doesnt handle the case where `self.writes_all` and `other.reads_all`

Reply:
What do you expect should happen?
If `self.writes_all`, that takes the first branch. If `other` accesses any element, there's a conflict.

An empty function (`|| {}`) doesn't access anything, so it shouldn't conflict.

Reply:
`read_all()` doesnt set `reads_and_writes` it jsut sets `reads_all` therefore `other.reads_and_writes.count_ones(..) == 0` is not sufficient to detect no reads. Same is true of `writes_all()` wouldnt this fail to detect if `other.writes_all` is true?

Reply:
Resolving in favor of centralizing discussion under my latest comment.